py2neo/util.py

```diff
@@ -33,6 +33,9 @@
 
 
 def snake_case(s):
+    if is_collection(s):
+        s = "_".join(s)
```

Comment:
This would export unexpected behaviour from the snake_case function. It would be better to perform joining _outside_ the function.